### PR TITLE
SDK: Use newly-merged notifications panel code in Calypso

### PR DESCRIPTION
**Pull request by dmsnell**
See #26524 and #26757

After importing the notifications panel code into Calypso this patch
makes the changes necessary for us to start importing that code directly
instead of through the `node_modules` import.

**Testing**

Run Calypso and verify that your notifications in the masterbar load, that
they look right visually, and that they don't throw any new errors.

### CODE REVIEW

**Review by vindl — APPROVED** [review]
I gave this a spin and everything worked as expected. Nice work @dmsnell!

File: bin/sdk/notifications.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+			},
+			output: {
+				path: outputDir,
+				filename: 'build.min.js',
+			},
+			plugins: [
+				...baseConfig.plugins,
+				new HtmlWebpackPlugin( {
+					filename: path.join( outputDir, 'root.html' ),
+					gitDescribe: spawnSync( 'git', [ 'describe', '--always', '--dirty', '--long' ], {
```
Comment (jsnajdr) [inline bin/sdk/notifications.js]: TIL about `git describe`! Will it do anything sensible though in the Calypso repo? Right now it shows `desktop/2.4.0-12909-ga8ca144600`

File: bin/sdk/notifications.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const HtmlWebpackPlugin = require( 'html-webpack-plugin' );
+const path = require( 'path' );
+const spawnSync = require( 'child_process' ).spawnSync;
+
+exports.config = ( { argv: { outputDir }, getBaseConfig, __rootDir } ) => {
+	const baseConfig = getBaseConfig();
+
+	return {
+		...baseConfig,
+		...{
```
Comment (jsnajdr) [inline bin/sdk/notifications.js]: nit: `...{` can be omitted, specify the extra properties directly.

**Review by jsnajdr — COMMENTED** [review]
Running `npm run sdk:notifications` fails with error:
```
$ npm run sdk:notifications

> wp-calypso@0.17.0 sdk:notifications /Users/jsnajdr/src/wp-calypso
> node bin/sdk-cli.js notifications

/Users/jsnajdr/src/wp-calypso/node_modules/yargs/yargs.js:1144
      else throw err
           ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at assertPath (path.js:39:11)
    at Object.join (path.js:1157:7)
    at Object.exports.config (/Users/jsnajdr/src/wp-calypso/bin/sdk/notifications.js:26:21)
    at build (/Users/jsnajdr/src/wp-calypso/bin/sdk-cli.js:46:24)
    at Object.handler (/Users/jsnajdr/src/wp-calypso/bin/sdk-cli.js:133:20)
    at Object.runCommand (/Users/jsnajdr/src/wp-calypso/node_modules/yargs/lib/command.js:237:44)
    at Object.parseArgs [as _parseArgs] (/Users/jsnajdr/src/wp-calypso/node_modules/yargs/yargs.js:1059:30)
    at Object.get [as argv] (/Users/jsnajdr/src/wp-calypso/node_modules/yargs/yargs.js:1000:21)
```
That doesn't change even if I specify the `--output-dir` option.

File: bin/sdk/notifications.js (hunk 1 above)
Comment (dmsnell) [reply bin/sdk/notifications.js]: we added this in after several issues related to the build environment. at worst it's an extra meta tag but at best it helps us quickly pinpoint variables that could have caused a broken build

File: bin/sdk/notifications.js (hunk 2 above)
Comment (dmsnell) [reply bin/sdk/notifications.js]: removed in 381a0563d6

**Comment by dmsnell** [comment]
> That doesn't change even if I specify the --output-dir option.

@jsnajdr I was able to reproduce if I omitted the `--output-dir` option but not if I supplied it. did you by any chance forget to include the extra `--` before the args?

```bash
npm run sdk:notifications -- --output-dir=/my/output/path
```

**Comment by jsnajdr** [comment]
> did you by any chance forget to include the extra -- before the args?

Of course, that's exactly what I did :smile: Now with `--` the notification panel is built successfully. Thanks!

**Review by jsnajdr — APPROVED** [review]
This patch looks ready to :ship: Thanks for working on it! :+1:

File: client/notifications/index.jsx [hunk 3]
```diff
@@ -28,7 +27,7 @@ import { connect } from 'react-redux';
 import analytics from 'lib/analytics';
 import config from 'config';
 import { recordTracksEvent } from 'state/analytics/actions';
-import NotificationsPanel, { refreshNotes } from 'notifications-panel';
+import NotificationsPanel, { refreshNotes } from './src/panel/Notifications';
```
Comment (blowery) [inline client/notifications/index.jsx]: is there a plan to clean up notifications usage of casing on dir names? iirc that's a pain to clean up on a mac...

File: webpack.config.js [hunk 4]
```diff
@@ -161,7 +161,7 @@ function getWebpackConfig( { cssFilename, externalizeWordPressPackages = false }
 			rules: [
 				{
 					test: /\.jsx?$/,
-					exclude: /node_modules[\/\\](?!notifications-panel)/,
+					exclude: /node_modules\//,
```
Comment (blowery) [inline webpack.config.js]: yay

**Pull request merged**